Badge: Fix the UI mismatch, add a new variant with tests

ritikamotwani opened the pull request:
Fixes: #192 
To check the changes: https://docs.google.com/document/d/1pHImZ2YXcG0WxMU5ll1y9zBygyErR2EjR8xu3Y2lgDw/edit

ritikamotwani committed:
Badge: Fix the UI mismatch, add a new variant with tests

tingtingdisco commented:
@ritikamotwani could you confirm with designers that do we still need `sup`? thank u

tingtingdisco commented on src/General/Badge/Badge.tsx:
```diff
@@ -26,6 +33,7 @@ const Badge: React.FunctionComponent<Props> = ({
 interface Props extends React.ComponentPropsWithoutRef<typeof BadgeContainer> {
   label: string | number;
   sup?: boolean;
+  type?: string;
```
should be `variant`?

ritikamotwani replied:
Yes yes, forgot to commit this file after changing the prop names to variant, from type.

tingtingdisco commented:
please update storybook page, to include the default type, and update the margin of the examples
![Screen Shot 2020-05-25 at 10 58 08 AM](https://user-images.githubusercontent.com/11376798/82774221-d11eaf00-9e76-11ea-8688-42f2af677186.png)
![Screen Shot 2020-05-25 at 10 58 39 AM](https://user-images.githubusercontent.com/11376798/82774224-d4199f80-9e76-11ea-806a-378df12e6c52.png)

ritikamotwani committed:
Badge: Update the variable name to variant

ritikamotwani committed:
Badge: Update the storybook page for the badge component to include all styles

ritikamotwani commented:
> please update storybook page, to include the default type, and update the margin of the examples

Updated the storybook with all default types, and added another div to show the accompanied margin.

tingtingdisco reviewed (approved):
LGTM
